Unable to link RISC-V linux-gnu and linux-musl executables using zig cc and CMake

### Zig Version

0.14.0-dev.2384+cbc05e0b1

### Steps to Reproduce and Observed Behavior

Simple toolchain file:
```cmake
set(CMAKE_SYSTEM_NAME "Linux")
set(CMAKE_SYSTEM_VERSION 1)
set(CMAKE_SYSTEM_PROCESSOR "riscv64")
set(CMAKE_CROSSCOMPILING TRUE)
#set(CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY)
set(CMAKE_C_COMPILER "zig" cc -target riscv64-linux-gnu)
set(CMAKE_CXX_COMPILER "zig" c++ -target riscv64-linux-gnu)
```
The try-static-library thing is just to avoid the compiler checks. The linking step fails, so we want the compiler check.

Simple build script:
```sh
mkdir -p .zig
pushd .zig
cmake .. -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE="$TOOLCHAIN"
make -j
popd
```
Compiler test:
```
$ bash zig.sh 
~/github/godot-sandbox-programs/.zig ~/github/godot-sandbox-programs
-- The C compiler identification is Clang 19.1.0
-- The CXX compiler identification is Clang 19.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /home/gonzo/zig/zig
-- Check for working C compiler: /home/gonzo/zig/zig - broken
CMake Error at /usr/share/cmake-3.22/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "/home/gonzo/zig/zig"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /home/gonzo/github/godot-sandbox-programs/.zig/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/gmake -f Makefile cmTC_aaa0d/fast && /usr/bin/gmake  -f CMakeFiles/cmTC_aaa0d.dir/build.make CMakeFiles/cmTC_aaa0d.dir/build
    gmake[1]: Entering directory '/home/gonzo/github/godot-sandbox-programs/.zig/CMakeFiles/CMakeTmp'
    Building C object CMakeFiles/cmTC_aaa0d.dir/testCCompiler.c.o
    /home/gonzo/zig/zig cc -target riscv64-linux-gnu    -MD -MT CMakeFiles/cmTC_aaa0d.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_aaa0d.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_aaa0d.dir/testCCompiler.c.o -c /home/gonzo/github/godot-sandbox-programs/.zig/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_aaa0d
    /usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_aaa0d.dir/link.txt --verbose=1
    /home/gonzo/zig/zig cc -target riscv64-linux-gnu CMakeFiles/cmTC_aaa0d.dir/testCCompiler.c.o -o cmTC_aaa0d   -latomic 
    error: unable to find dynamic system library 'atomic' using strategy 'paths_first'. searched paths: none
    gmake[1]: *** [CMakeFiles/cmTC_aaa0d.dir/build.make:100: cmTC_aaa0d] Error 1
    gmake[1]: Leaving directory '/home/gonzo/github/godot-sandbox-programs/.zig/CMakeFiles/CMakeTmp'
    gmake: *** [Makefile:127: cmTC_aaa0d/fast] Error 2
```

It's erroneously trying to link against libatomic: `-latomic`. Same behavior on both linux-gnu and linux-musl.

As a sanity check I tried to compile manually on the command-line, and that works as -latomic is not being added. But, maybe this is a CMake bug? And is -latomic supposed to be present? Can it be an empty library?

### Expected Behavior

Don't link against libatomic.

--- a/lib/std/zig/target.zig
+++ b/lib/std/zig/target.zig
@@ -263,7 +263,8 @@ pub fn isLibCxxLibName(target: std.Target, name: []const u8) bool {
 
     return eqlIgnoreCase(ignore_case, name, "c++") or
         eqlIgnoreCase(ignore_case, name, "stdc++") or
-        eqlIgnoreCase(ignore_case, name, "c++abi");
+        eqlIgnoreCase(ignore_case, name, "c++abi") or
+        eqlIgnoreCase(ignore_case, name, "supc++");
 }
 
 fn eqlIgnoreCase(ignore_case: bool, a: []const u8, b: []const u8) bool {

--- a/src/main.zig
+++ b/src/main.zig
@@ -3838,10 +3838,7 @@ fn createModule(
                         create_module.opts.link_libunwind = true;
                         continue;
                     },
-                    .only_compiler_rt => {
-                        warn("ignoring superfluous library '{s}': this dependency is fulfilled instead by compiler-rt which zig unconditionally provides", .{lib_name});
-                        continue;
-                    },
+                    .only_compiler_rt => continue,
                 }
 
                 if (target.isMinGW()) {

--- a/src/main.zig
+++ b/src/main.zig
@@ -3832,7 +3832,7 @@ fn createModule(
                     create_module.opts.link_libcpp = true;
                     continue;
                 }
-                switch (target_util.classifyCompilerRtLibName(target, lib_name)) {
+                switch (target_util.classifyCompilerRtLibName(lib_name)) {
                     .none => {},
                     .only_libunwind, .both => {
                         create_module.opts.link_libunwind = true;

--- a/src/target.zig
+++ b/src/target.zig
@@ -272,7 +272,9 @@ pub fn classifyCompilerRtLibName(name: []const u8) CompilerRtClassification {
     {
         return .only_compiler_rt;
     }
-    if (std.mem.eql(u8, name, "unwind")) {
+    if (std.mem.eql(u8, name, "unwind") or
+        std.mem.eql(u8, name, "gcc_eh"))
+    {
         return .only_libunwind;
     }
     return .none;

--- a/src/target.zig
+++ b/src/target.zig
@@ -269,7 +269,8 @@ pub fn classifyCompilerRtLibName(name: []const u8) CompilerRtClassification {
     }
     if (std.mem.eql(u8, name, "compiler_rt") or
         std.mem.eql(u8, name, "gcc") or
-        std.mem.eql(u8, name, "atomic"))
+        std.mem.eql(u8, name, "atomic") or
+        std.mem.eql(u8, name, "ssp"))
     {
         return .only_compiler_rt;
     }

--- a/src/target.zig
+++ b/src/target.zig
@@ -268,6 +268,7 @@ pub fn classifyCompilerRtLibName(name: []const u8) CompilerRtClassification {
         return .both;
     }
     if (std.mem.eql(u8, name, "compiler_rt") or
+        std.mem.eql(u8, name, "gcc") or
         std.mem.eql(u8, name, "atomic"))
     {
         return .only_compiler_rt;

--- a/src/target.zig
+++ b/src/target.zig
@@ -260,8 +260,8 @@ pub fn supportsReturnAddress(target: std.Target) bool {
 
 pub const CompilerRtClassification = enum { none, only_compiler_rt, only_libunwind, both };
 
-pub fn classifyCompilerRtLibName(target: std.Target, name: []const u8) CompilerRtClassification {
-    if (target.abi.isGnu() and std.mem.eql(u8, name, "gcc_s")) {
+pub fn classifyCompilerRtLibName(name: []const u8) CompilerRtClassification {
+    if (std.mem.eql(u8, name, "gcc_s")) {
         // libgcc_s includes exception handling functions, so if linking this library
         // is requested, zig needs to instead link libunwind. Otherwise we end up with
         // the linker unable to find `_Unwind_RaiseException` and other related symbols.

--- a/src/target.zig
+++ b/src/target.zig
@@ -267,7 +267,9 @@ pub fn classifyCompilerRtLibName(target: std.Target, name: []const u8) CompilerR
         // the linker unable to find `_Unwind_RaiseException` and other related symbols.
         return .both;
     }
-    if (std.mem.eql(u8, name, "compiler_rt")) {
+    if (std.mem.eql(u8, name, "compiler_rt") or
+        std.mem.eql(u8, name, "atomic"))
+    {
         return .only_compiler_rt;
     }
     if (std.mem.eql(u8, name, "unwind")) {

